tests/aequilibrae/project/test_project_triggers.py

```diff
@@ -0,0 +1,191 @@
+from unittest import TestCase
+import tempfile
+import os
+from warnings import warn
+from shutil import copytree, rmtree
+from aequilibrae.project import Project
+import uuid
+import sqlite3
+from ...data import no_triggers_project
+
+
+class TestProject(TestCase):
+    def setUp(self) -> None:
+        self.temp_proj_folder = os.path.join(tempfile.gettempdir(), uuid.uuid4().hex)
+        copytree(no_triggers_project, self.temp_proj_folder)
+        self.proj = Project()
+        self.proj.open(self.temp_proj_folder)
+        self.curr = self.proj.conn.cursor()
+
+    def tearDown(self) -> None:
+        self.proj.close()
+        rmtree(self.temp_proj_folder)
+
+    def test_link_type_triggers(self):
+        root = os.path.dirname(os.path.realpath(__file__)).replace('tests', '')
+        qry_file = os.path.join(root, "network/database_triggers/link_type_table_triggers.sql")
+        with open(qry_file, "r") as sql_file:
+            query_list = sql_file.read()
+            query_list = [cmd for cmd in query_list.split("#")]
+
+            def reboot_cursor():
+                self.proj.conn.commit()
+                self.curr = self.proj.conn.cursor()
+
+        for cmd in query_list:
+            if 'link_type_single_letter_update' in cmd:
+                sql = "UPDATE 'link_types' SET link_type_id= 'ttt' where link_type_id='t'"
+                self.curr.execute(sql)
+
+                self.curr.execute(cmd)
+                reboot_cursor()
+
+                with self.assertRaises(sqlite3.IntegrityError):
+                    sql = "UPDATE 'link_types' SET link_type_id= 'ww' where link_type_id='w'"
+                    self.curr.execute(sql)
+                reboot_cursor()
+
+            elif 'link_type_single_letter_insert' in cmd:
```

Comment:
I'm not sure I like this setup, you seem to be running through a list of sql commands from file and based on the content of those, you are running some other sql first to set things up for the test. 

Why not just embed all the sql here in the test instead of having it in two places.

Reply:
Agreed.  I changed how it works for the modes and link_types tables yesterday. I will do the same for these

Reply:
You can check it how I did it here:
https://github.com/AequilibraE/aequilibrae/pull/150/files?file-filters%5B%5D=.py&file-filters%5B%5D=.rst&file-filters%5B%5D=.sql&file-filters%5B%5D=.sqlite&file-filters%5B%5D=.txt#diff-bad12097672754322ad36a32502d7b54